_posts/2021-06-09-ejb-over-http.adoc

```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: "Authenticating EJBs over HTTP"
+date: 2021-06-09
+tags: wildfly elytron
+synopsis: An overview of the how to setup EJBs to operate over HTTP.
+author: araskar
+---
+
+The *https://github.com/wildfly-security-incubator/elytron-examples/tree/master/ejb-http[ejb-http]* quickstart will describe the steps to setting up EJBs using HTTP as the transport. It will also secure it with a FileSystem realm, and use a credential store to access the EJB. This will focus on the configuration of the server to allow it to communicate over HTTP.
+
+== Configure the filesystem realm
+First we will connect to the server via 
+
+[source, shell]
+----
+$ WILDFLY_HOME/bin/jboss-cli.sh --connect
+----
+`WILDFLY_HOME` refers to the location where Wildfly is installed. For example it could be something like `~/wildfly/build/target/wildfly-24.0.0.Beta1-SNAPSHOT`
+
+Then we will create a file system realm called fsRealm, and add a directory in the jboss server config directory called fs-realm-users that will contain all the data of the identity of each user created
+
+[source]
+----
+/subsystem=elytron/filesystem-realm=fsRealm:add(path=fs-realm-users, relative-to=jboss.server.config.dir)
+----
+
+Then we will create a new user, assign them a password, and give them the role of guest
+[source]
+----
+/subsystem=elytron/filesystem-realm=fsRealm:add-identity(identity=quickstartUser)
+/subsystem=elytron/filesystem-realm=fsRealm:set-password(clear={password=quickstartPwd1!},identity=quickstartUser)
+/subsystem=elytron/filesystem-realm=fsRealm:add-identity-attribute(identity=quickstartUser, name=Roles, value=[guest])
+----
+
+We will also add a role decoder to be able to process the roles assigned
+
+[source]
+----
+/subsystem=elytron/simple-role-decoder=from-roles-attribute:add(attribute=Roles)
+----
+
+== Configure the Security Domain
+
+We will create a security domain that resides in the filesystem realm and will use the role decoder we created above
+
+[source]
+----
+/subsystem=elytron/security-domain=fsSD:add(realms=[{realm=fsRealm, role-decoder=from-roles-attribute}], default-realm=fsRealm,permission-mapper=default-permission-mapper)
+----
+
+We will also add the filesystem security domain mapping in the EJB3 subsystem so that the server uses the correct security domain 
+
+[source]
+----
+/subsystem=ejb3/application-security-domain=other:add(security-domain=fsSD)
+----
+
+== Configure the HTTP Transport
+
+First we must configure the http-authentication-factory for the server to authenticate over HTTP.
+Here we set the mechanism to BASIC (but could also be set to FORM), point to the fsSD security-domain, and the fsRealm filesystem-realm, and we call the newly created http-authentication-factory, example-http-auth
```

Comment:
"(but could also be set to FORM)" ->

"(but could also be set to another HTTP mechanism, like FORM for example)"

Reply:
Fixed